### ISSUE TITLE
how to require at least one space between 2 tokens?

### ISSUE THREAD

**Issue by shi-yan** [pest-parser/pest#502]
I want to match

"aaa bbb"

there must be at least one space between the two tokens "aaa" and "bbb"

the following works,

```
WHITESPACE = _{ " " | "\t" }
accept = ${"aaa" ~ WHITESPACE+  ~ "bbb" ~ NEWLINE}
```

but is very inconvenient, because $ disables implicit whitespaces. So I have to manually add whitespace+ between all tokens. (this is fine, but I also need to add comments /**/ , as according to the doc, $ disables both )

 
Then I tried this,

```
WHITESPACE = _{ " " | "\t" }
accept = {"aaa"  ~ "bbb" ~ NEWLINE}
```

this works too, however it also matches "aaabbb". I want to make sure there is at least one space.

In my opinion, the correct way should be
```
WHITESPACE = _{ " " | "\t" }
accept = {"aaa" ~ WHITESPACE  ~ "bbb" ~ NEWLINE}
```

I thought this would enforce at least one space between "aaa" and "bbb". 

however I got parsing error:

```
let pairs = IdentParser::parse(Rule::accept, "aaa bbb
").unwrap_or_else(|e| panic!("{}", e));
```

```
thread 'main' panicked at ' --> 1:1
  |
1 | aaa bbb␊
  | ^---
  |
  = expected accept', src/main.rs:29:23
```

This I don't understand?

**Comment by dragazo** [comment]
@siberianbluerobin I just tried a simplified version of the last grammar in the editor and it fails to parse even when there *is* whitespace.

```
WHITESPACE = _{ " " | "\t" }
accept = { "aaa" ~ WHITESPACE ~ "bbb"}
```

```
aaa bbb
```

```
 --> 1:1
  |
1 | aaa bbb
  | ^---
  |
  = expected accept
```

I just found out about pest a few days ago. I'd like to use it for my parsing needs, but I need this to at least be possible...

**Comment by siberianbluerobin** [comment]
@dragazo My mistake, you are right, when I tested the grammar I didn't copy and paste just write short names for the identifiers

This works
```
MYWHITESPACE = _{ " " | "\t" }
accept = { "aaa" ~ MYWHITESPACE ~ "bbb"}
```

Sound like the WHITESPACE rule isn't replaced

**Comment by dragazo** [comment]
@siberianbluerobin Yeah, by not defining `WHITESPACE` that's basically atomic, which should work. I think I'll just start making normal non-atomic rules for most things and have atomic helpers for special cases like this. A bit annoying, but it'll do :/ Just another level of unwrapping.

**Comment by max-sixty** [comment]
I'm facing a similar situation. 

Assuming we can't make rules both non-atomic, and silent, something that would be helpful for this is a rule type which is atomic only for that rule — no inheritance.

So:
```
accept = #{"aaa" ~ WHITESPACE+  ~ other_rule ~ NEWLINE}
```

...forces whitespace between `aaa` and whatever `other_rule` is, without disallowing whitespace in `other_rule`, and without having to make `other_rule` non-atomic.

If `other_rule` could be _both_ non-atomic and silent, that would also be fine — it would there's lots of non-atomic rules as @dragazo suggests, but that's a fairly small cost.
